### ISSUE TITLE
OOM performing search

### ISSUE THREAD

**Issue by stroomdev10** [gchq/stroom#1032]
```
ERROR  [2018-12-14T10:12:29.334Z] [Search Result Sender #13 - GenericServerTask] stroom.cluster.server.ClusterCallServiceRemote -  Requested array size exceeds VM limit
java.lang.OutOfMemoryError: Requested array size exceeds VM limit
        at java.util.Arrays.copyOf(Arrays.java:3236)
        at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
        at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
        at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:135)
        at sun.net.www.http.PosterOutputStream.write(PosterOutputStream.java:63)
        at com.caucho.hessian.io.HessianOutput.printString(HessianOutput.java:922)
        at com.caucho.hessian.io.HessianOutput.printLenString(HessianOutput.java:895)
        at com.caucho.hessian.io.HessianOutput.writeMapBegin(HessianOutput.java:382)
        at com.caucho.hessian.io.AbstractHessianOutput.writeObjectBegin(AbstractHessianOutput.java:538)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:167)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.ArraySerializer.writeObject(ArraySerializer.java:69)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.CollectionSerializer.writeObject(CollectionSerializer.java:117)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.MapSerializer.writeObject(MapSerializer.java:116)
ERROR  [2018-12-14T10:12:29.336Z] [Search Result Sender #13 - GenericServerTask] stroom.task.cluster.ClusterWorkerImpl -  Requested array size exceeds VM limit
java.lang.OutOfMemoryError: Requested array size exceeds VM limit
        at java.util.Arrays.copyOf(Arrays.java:3236)
        at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
        at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
        at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:135)
        at sun.net.www.http.PosterOutputStream.write(PosterOutputStream.java:63)
        at com.caucho.hessian.io.HessianOutput.printString(HessianOutput.java:922)
        at com.caucho.hessian.io.HessianOutput.printLenString(HessianOutput.java:895)
        at com.caucho.hessian.io.HessianOutput.writeMapBegin(HessianOutput.java:382)
        at com.caucho.hessian.io.AbstractHessianOutput.writeObjectBegin(AbstractHessianOutput.java:538)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:167)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.ArraySerializer.writeObject(ArraySerializer.java:69)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.CollectionSerializer.writeObject(CollectionSerializer.java:117)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.UnsafeSerializer$ObjectFieldSerializer.serialize(UnsafeSerializer.java:297)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject10(UnsafeSerializer.java:190)
        at com.caucho.hessian.io.UnsafeSerializer.writeObject(UnsafeSerializer.java:178)
        at com.caucho.hessian.io.HessianOutput.writeObject(HessianOutput.java:324)
        at com.caucho.hessian.io.MapSerializer.writeObject(MapSerializer.java:116)
```
text file of above stack trace
[20181214.txt](https://github.com/gchq/stroom/files/2680011/20181214.txt)

**Comment by burnalting** [comment]
Has anyone a workaround for this condition. It occurs in V5 as well and I am looking at ways of circumventing the problem if possible.

**Comment by stroomdev66** [comment]
Fixed in 7.0
